src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs

```diff
@@ -130,17 +135,22 @@ public static long GetTimestamp()
         // Get the elapsed ticks.
         private long GetRawElapsedTicks()
         {
-            long timeElapsed = _elapsed;
+            long stateCopy = _state;
 
-            if (_isRunning)
+            // Unstarted, or stopped with an elapsed time > 0 .
+            if (stateCopy <= 0)
             {
-                // If the Stopwatch is running, add elapsed time since
-                // the Stopwatch is started last time.
-                long currentTimeStamp = GetTimestamp();
-                long elapsedUntilNow = currentTimeStamp - _startTimeStamp;
-                timeElapsed += elapsedUntilNow;
+                return Math.Abs(stateCopy);
```

Comment:
Optimization point: there are other places that sets the highest bit to 1, instead of negating the number. Generally, setting highest bit could potentially be faster than negation, and clearing highest bit is definitely faster than `Abs` (`Abs` is several times more costful than bit operations, and has a branch).

Reply:
I played around with a couple different ways to accomplish this, a branch ends up being the best in my testing.